include/cpp2util.h

```diff
@@ -1433,50 +1578,59 @@ auto is( std::variant<Ts...> const& x, C const& value ) {
 
 template<typename T, typename... Ts>
 auto is( std::variant<Ts...> const& x ) {
+        //std::cout << __PRETTY_FUNCTION__ << " " << __LINE__ << std::endl;
     return type_find_if<Ts...>([&]<typename It>(It const&) -> bool {
         if constexpr (std::is_same_v< typename It::type, T >) { return x.index() == It::index; } else { return false; }
     }) != std::variant_npos;
 }
 
 template<std::same_as<empty> T, typename... Ts>
 auto is( std::variant<Ts...> const& x ) {
+        //std::cout << __PRETTY_FUNCTION__ << " " << __LINE__ << std::endl;
     if (x.valueless_by_exception()) 
         return true;
     if constexpr (is_any<std::monostate, Ts...>) 
         return std::get_if<std::monostate>(&x) != nullptr;
     return false;
 }
 
-template<typename T, typename... Ts>
-auto as( std::variant<Ts...> && x ) -> decltype(auto) {
-    T* ptr = nullptr;
-    type_find_if<Ts...>([&]<typename It>(It const&) -> bool {
-        if constexpr (std::is_same_v< typename It::type, T >) { if (x.index() ==  It::index) { ptr = &std::get<It::index>(x); return true; } }; return false;
-    });
-    if (!ptr) { Throw( std::bad_variant_access(), "'as' cast failed for 'variant'"); } 
-    return *ptr;
-}
+template <typename X, template<typename...> class C>
+concept specialization_of_template = requires (std::remove_cvref_t<X> x) {
+    { is<C>(std::forward<X>(x)) } -> std::same_as<std::true_type>;
```

Comment:
```suggestion
concept specialization_of_template = requires {
    { is<C>(std::declval<X>()) } -> std::same_as<std::true_type>;
```

Reply:
most probably `std::remove_cvref_t<X>` and `std::forward<X>(x)` are not needed. I have been wondering what is better: use `std::declval<X>()` or `requires(X x) { ... }`. The second option is more compelling to me (especially after writing many concepts.

I am continue to clean this code up.